src/main.c

```diff
@@ -534,11 +534,29 @@ void dict(struct state *s)
 			}
 		} else if (strcmp(subfunc, "auth") == 0) {
 			if (peek_arg(&(s->cmd_list)) == NULL) {
-				printf("No username specified.\n");
-			} else {
+				char filename[] = "user.inf";
+				FILE*f;
+				int len;
+				rsp = exword_get_file(s->device, basename(filename), &buffer, &len);
+				if (rsp == 0x20)
+					rsp = write_file(filename, buffer, len);
+				free(buffer);
+				if ((f = fopen("user.inf", "r")) == NULL){
+					user = NULL;
+					printf("No username specified.\n");
+				}
+				else{
+					user = xmalloc(32);
+					fscanf(f, "%s\n", user);
```

Comment:
Use "%31s\n" for this format specifier to ensure the buffer can't be overrun here.